posix/common

```diff
@@ -12,6 +12,11 @@ if [ ! -d .git ]; then
   git config --global --add safe.directory '*'
   git remote add origin ${DRONE_REMOTE_URL}
   set +x
+else
+  set -x
+  git config --global --add safe.directory '*'
+  git remote set-url origin ${DRONE_REMOTE_URL}
```

Comment:
let's test it in qa once before merging.

Reply:
asking this because we also run git clone step in containerless mode on cloud infra, not sure how that will be impacted

Reply:
@devkimittal I've ran automations for this change in using delegate + local harness-docker-runner. I've shared the automation results in the PR's descriptions. Basically, with these changes, there is one automation case in which the pipeline succeeds instead of fails, due to the new `git remote set-url origin ${DRONE_REMOTE_URL}`. It seem acceptable and harmless to me. Please check the details in the PR's description.

Can you help getting this PR merged? We will then deploy this to QA for testing the hosted use-case, since hosted automation doesn't work in Devspace.